apps/src/blockly/customBlocks/googleBlockly/mutators/behaviorDefMutator.js

```diff
@@ -80,9 +84,8 @@ export const behaviorDefMutator = {
     const state = Object.create(null);
     state['procedureId'] = this.getProcedureModel().getId();
     state['behaviorId'] = this.behaviorId;
-    if (this.description) {
-      state['description'] = this.description;
-    }
+
+    handleSaveDescription(this, state);
```

Comment:
Could we match the existing pattern by making these functions return the appropriate description and doing the assignment `saveExtraState`?
```
state['description'] = getDescriptionFromBlock(this);
```
My gut feeling is that approach makes it clearer/more straightforward to someone what's actually happening in that function and might make unit testing easier because the helper isn't modifying the state.

Reply:
Second question: Should we keep a conditional check here for whether description is truthy?

Reply:
To the first question: yes that makes sense, will update.
To the second question: it doesn't really matter either way. In the load step, even if the description is undefined, if we have a modal function editor we will still want to create the description field so the user can add a description. So it feels ok to me to set the description to undefined/empty string here.